### PR TITLE
additional parameters for hf

### PR DESCRIPTION
**Pull request by JulianneKnott**
support more of hf's generation params in the fastseq cli

### CODE REVIEW

**Review by JiushengChen — APPROVED** [review]

File: fastseq_cli/transformers_generate.py [hunk 1]
```diff
@@ -354,7 +368,12 @@ def run_generate():
                         default=None, required=False)
     parser.add_argument("--max_gen_length", type=int,
                         help="max length for generation",
-                        default=None, required=False)
+                        default=None, required=False),
+    parser.add_argument("--max_new_tokens", type=int,
```
Comment (feihugis) [inline fastseq_cli/transformers_generate.py:372]: If `max_gen_length` and `max_new_tokens` serve the same purpose, could we just use one of them?

Comment (JulianneKnott) [reply fastseq_cli/transformers_generate.py:372]: both exist already in hf. This change just makes the existing functionality available in fastseq.  https://github.com/huggingface/transformers/blob/3f76bf54ff3754f9f6639ee8f9c36775be9b8ce6/src/transformers/generation_utils.py#L829

File: fastseq_cli/transformers_generate.py [hunk 2]
```diff
@@ -368,6 +387,18 @@ def run_generate():
...
+                        help="id fo the end-of-sequence token")
+    parser.add_argument("--temperature", type=float,
+                        default=None, required=False)
+    parser.add_argument("--top_k", type=int,
+                        default=None, required=False)
+    parser.add_argument("--top_p", type=float, 
+                        default=None, required=False)
+    parser.add_argument("--repetition_penalty", type=float,
+                        default=None, required=False)
+    parser.add_argument("--do_sample", action="store_true")
```
Comment (feihugis) [inline fastseq_cli/transformers_generate.py:401]: Could you add `help` doc for these variables?

File: fastseq_cli/transformers_generate.py (hunk 1 above)
Comment (feihugis) [reply fastseq_cli/transformers_generate.py:372]: Gotcha.

**Review by feihugis — APPROVED** [review]

**Pull request merged**